mempool/txpool/legacypool/legacypool.go

```diff
@@ -1900,3 +1967,23 @@ func (pool *LegacyPool) Clear() {
 func (pool *LegacyPool) HasPendingAuth(addr common.Address) bool {
 	return pool.all.hasAuth(addr)
 }
+
+// CurrentCacheContext gets a cached version of the current context in the
+// pools stateDB. Note that this may momentarily differ from the BlockChain's
+// context.
+func (pool *LegacyPool) CurrentCacheContext() sdk.Context {
```

Comment:
I don't really love introducing SDK types into the geth code as this makes it harder to maintain. 

Is there a way where we can cache the current query context in mempool.go and use that inside the callback code instead of passing it in as a parameter from legacypool?